### PR TITLE
ikuz

### PR DESCRIPTION
**Pull request by seankalejs**
Added Tr Quantity (without vehicle comparison for now)

### CODE REVIEW

File: lib/include/customer.h [hunk 1]
```diff
@@ -3,13 +3,15 @@
 #include <cstdint>
 #include <utility>
 #include <vector>
+#include <transportation_quantity.h>
```
Comment (andrgolubev) [inline lib/include/customer.h]: please, use this style for our headers: `#include "transportation_quantity.h"`

File: lib/include/transportation_quantity.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <cstdint>
+
+namespace vrp {
+/// Transportation Quantity representation
+class TrQuant {
```
Comment (andrgolubev) [inline lib/include/transportation_quantity.h]: let's use the full name instead: `TransportationQuantity`

File: lib/include/customer.h [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
 
 namespace vrp {
 /// Customer representation
 class Customer {
 public:
     int id = 0; ///< customer id. 0 is reserved for depot
     int demand = 0;    ///< customer demand
+    TrQuant trquant_c; ///customer demand quantity (of 1 item)
```
Comment (andrgolubev) [inline lib/include/customer.h]: i'd just name the member variable `quantity`

Comment (andrgolubev) [reply lib/include/customer.h]: by the way, this is not a quantity **for 1 item**. This is a **total** quantity needed by customer (as far as I understand)

File: lib/include/transportation_quantity.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <cstdint>
```
Comment (andrgolubev) [inline lib/include/transportation_quantity.h]: not needed

File: lib/include/vehicle.h [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+#include "transportation_quantity.h"
 
 namespace vrp {
 /// Vehicle representation
 class Vehicle {
 public:
     int id = 0;  ///< vehicle id
-    int capacity = 0;  ///< vehicle capacity
-    int max_weight = 0;  ///< max vehicle weight
+    TrQuant trquant_v; ///vehicle quantity
```
Comment (andrgolubev) [inline lib/include/vehicle.h]: I'd simply name `quantity`

File: test_data/sample_input.csv [hunk 6]
```diff
@@ -1,12 +1,12 @@
 table customer
-id; demand; hard_tw_begin; hard_tw_end; soft_tw_begin; soft_tw_end; service_time; suitable_vehicles
-1; 10; 0; 200; 50; 70; 10; 0; 1; 2
-2; 5; 10; 50; 10; 20; 5; 0; 3
-3; 1; 100; 110; 105; 110; 1; 1
-4; 2; 0; 20; 5; 15; 3; 2; 3
+id; demand; volume; weight; hard_tw_begin; hard_tw_end; soft_tw_begin; soft_tw_end; service_time; suitable_vehicles
```
Comment (andrgolubev) [inline test_data/sample_input.csv]: please, remove demand field. demand `<=>` transportation quantity

**Review by andrgolubev — CHANGES_REQUESTED** [review]

**Comment by andrgolubev** [comment]
Please, fix the comments and we'll merge the code

File: lib/include/customer.h (hunk 3 above)
Comment (batsyn) [reply lib/include/customer.h]: There should be no "int demand".
Instead it should be "TransportationQuantity demand".
Yes, it is total quantity.

File: lib/include/vehicle.h (hunk 5 above)
Comment (batsyn) [reply lib/include/vehicle.h]: It should be: TransportationQuantity capacity
And fields capacity and max_weight should be removed

File: test_data/sample_input.csv (hunk 6 above)
Comment (batsyn) [reply test_data/sample_input.csv]: yes, volume and weight determine the demand

File: lib/include/vehicle.h (hunk 5 above)
Comment (andrgolubev) [reply lib/include/vehicle.h]: agree with Mike

File: lib/include/customer.h (hunk 3 above)
Comment (andrgolubev) [reply lib/include/customer.h]: agree with Mike

**Comment by seankalejs** [comment]
I hope I understood correctly all the required changes :)

File: lib/include/transportation_quantity.h [hunk 7]
```diff
@@ -0,0 +1,8 @@
+namespace vrp {
```
Comment (andrgolubev) [inline lib/include/transportation_quantity.h:3]: you need `#pragma once` at the top

**Review by andrgolubev — CHANGES_REQUESTED** [review]
See my comment about `lib/include/transportation_quantity.h`. The rest is fine

**Comment by seankalejs** [comment]
Fixed

**Review by andrgolubev — APPROVED** [review]

**Pull request merged**